pdu/UDH.go

```diff
@@ -45,37 +52,24 @@ func (u *UDH) MarshalBinary() (b []byte, err error) {
 // The function returns the number of bytes read from src, and any error
 // detected while unmarshalling.
 func (u *UDH) UnmarshalBinary(src []byte) (int, error) {
-	if len(src) < 1 {
-		return 0, fmt.Errorf("Decode error UDHL %d underflow", 0)
+	if length := len(src); length > 1 && length < int(src[0]+1) {
```

Comment:
@linxGnu checked here.

Reply:
This check is wrong, if `src[0] == 255` then it's overflow after + operator. In this case, the check is invalid!
PTAL: https://play.golang.org/p/i97kByRfwbj